sdk/state/asset.go

```diff
@@ -41,3 +41,10 @@ func (a Asset) StringCanonical() string {
 	}
 	return fmt.Sprintf("%s:%s", a.Code(), a.Issuer())
 }
+
+func (a Asset) EqualsTrustLineAsset(ta xdr.TrustLineAsset) bool {
+	if ta.Type == xdr.AssetTypeAssetTypePoolShare {
+		return false
+	}
+	return ta.ToAsset().StringCanonical() == a.StringCanonical()
```

Comment:
💡 I'd flip this so as to compare the asset types we know about, rather than attempt to exclude all possible asset types we don't know about, since that set of asset types could get larger over time and it keeps this function scoped to the things it knows about.
```suggestion
	switch ta.Type {
	case xdr.AssetTypeNative, xdr.AssetTypeAssetTypeCreditAlphanum4, xdr. AssetTypeAssetTypeCreditAlphanum12:
		return ta.ToAsset().StringCanonical() == a.StringCanonical()
	default:
		return false
	}
```

Reply:
i was leaning towards simple, but agreed this does seem safer long term
https://github.com/stellar/experimental-payment-channels/pull/382/commits/323404c7a23c31714c22823246c502972d351df9